daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -285,6 +285,19 @@ sealed abstract class HasTxNodes {
       }
     )
 
+  def byInterfaceNodes: Iterable[Node.Action] =
+    nodes.values.flatMap(node =>
+      node match {
+        case action: Node.Action =>
+          if (action.byInterface.isDefined)
+            Iterator(action)
+          else
+            Iterator()
+        case _: Node.Rollback =>
+          Iterator()
+      }
+    )
+
```

Comment:
Not a very big deal, by I would prefer the node order to be preserved.
```suggestion
 def byInterfaceNodes: Iterable[Node.Action] = {
    fold(List.newBuilder[Node.Action]){
      case (acc, (_, node: Node.Action)) if node.byInterface.isDefined =>
        acc += node
      case (acc, _) => acc
    }
```

Reply:
In fact If you make this public, it is probably a good idea to keep the `NodeId`.
Maybe simpler to make it `lf` private.

Reply:
I'll make `byInterfaceNodes` private and clean it up, based on your suggestion.

Reply:
Here: #11672